webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java

```diff
@@ -138,14 +138,28 @@ private void verifyApacheCXFModuleDependencyRequirement(DeploymentUnit unit) {
         }
     }
 
-    private static boolean hasCxfModuleDependency(DeploymentUnit unit) {
+    static boolean hasCxfModuleDependency(DeploymentUnit unit) {
         final ModuleSpecification moduleSpec = unit.getAttachment(Attachments.MODULE_SPECIFICATION);
         for (ModuleDependency dep : moduleSpec.getUserDependencies()) {
             final String id = dep.getIdentifier().getName();
             if (cxfExportingModules.contains(id)) {
                 return true;
             }
         }
+        return hasParentCxfModuleDependency(unit);
+    }
+
+    private static boolean hasParentCxfModuleDependency(DeploymentUnit unit) {
+        if (unit.getParent() != null) {
+            final ModuleSpecification moduleSpec = unit.getParent().getAttachment(Attachments.MODULE_SPECIFICATION);
+            for (ModuleDependency dep : moduleSpec.getUserDependencies()) {
+                final String id = dep.getIdentifier().getName();
+                if (cxfExportingModules.contains(id) && dep.isExport()) {
+                    return true;
+                }
+            }
+            return hasParentCxfModuleDependency(unit.getParent());
```

Comment:
If I understand this correctly, here we already checked Ear deployment. Do we really need this recursion after this check ?  

I don't know if this case has already been checked:  there is another sub-deployment has cxf dependencies, and ear-subdeployments-isolated set to false . Do we need to look at this ?

Reply:
Agree, based on jboss deployment structure XSD, recursion is not necessary.

I added another check that should take `subdeployments-isolated` into account.